src/modules/form/util/formUtil.ts

```diff
@@ -567,6 +567,17 @@ export const autofillValues = ({
     // Stop iterating through autofill values
     return true;
   });
+
+  // For readonly items, if no autofill condition matched, clear out the item's value.
+  // Don't make any changes if the whole form is in viewOnly mode, though, because
+  // in that case this is being called on form load (not on change of an autofill condition),
+  // so we don't want to clear any form values.
+  if (!autofilled && item.readOnly && !viewOnly) {
+    values[item.linkId] = undefined;
+    return true;
+  }
```

Comment:
I think the "read only items" needs to be defined as `item.readOnly || item.type == 'display', unfortunately.. We have items like the below-- can you add something like this to the test form to ensure it works as expected?

```
                {
                  "text": "<b>Duration</b>: ${FORMAT_DURATION(value)}",
                  "type": "DISPLAY",
                  "link_id": "ServiceCode_1_duration",
                  "autofill_values": [
                    {
                      "formula": "ABS(ServiceCode_1_start - ServiceCode_1_end)",
                      "autofill_when": [
                        {
                          "operator": "EXISTS",
                          "question": "ServiceCode_1_start",
                          "answer_boolean": true
                        },
                        {
                          "operator": "EXISTS",
                          "question": "ServiceCode_1_end",
                          "answer_boolean": true
                        }
                      ],
                      "autofill_behavior": "ALL",
                      "autofill_readonly": true
                    }
                  ]
                }
```

Reply:
Yes, I added this, and with your suggested change it appears to be behaving as we expect on editable forms. I think this might be exposing an unrelated bug with the duration component on view-only forms but I have to dig a little more to confirm.

Reply:
Ok, I haven't tracked down the reason for the bug but to reproduce:
- Visit https://qa-hmis.openpath.host/admin/forms/marthas_custom_assessment/165/preview-draft
- Type something into the "Autofill when" field (Group 1)
- Type something into the "just a simple text box" field (Group 3)
- Switch to Read Only View
- Why does "Autofill when" retain its value, but "just a simple text box" does not??

Reply:
Oh, that's a good call out. The items that don't have `mapping`s don't retain their values when switching between edit/view modes, because the `FormPreview` component does the transformation process that changes the form state from `keyed by link id => keyed by field name => keyed by link id` when switching modes. We could probably get it working without that extra transformation, which would resolve this issue. I'll make a ticket. Thanks for finding!

Reply:
https://github.com/open-path/Green-River/issues/6487

Reply:
Ahaa, that makes sense, thanks for the sleuthing!